textile/chat-features.textile

```diff
@@ -28,6 +28,14 @@ h2(#general). General Principles
 * @(CHA-GP3)@ Wherever possible, Chat features should be exposed in the public API as properties of their parent. For example, @messages@ would be considered a property of a @room@ object. This allows for greater composability and extensibility in the future.
 * @(CHA-GP4)@ Avoid overloading methods and optional parameters. Prefer object-type parameters wherever practical and idiomatic.
 
+h2(#initialization). Initialization
```

Comment:
We also have a special requirement for React. In this case, the `ChatClientProvider` automatically updates the underlying Realtime client to set chat-react as the agent. We have do to this "post-constructor", as its not known at CTOR time whether the client's going to be used for React or not.

Is there a proposal for how this could be achieved?

Reply:
Not currently; I wasn't aware of this requirement. Let's discuss in https://github.com/ably/specification/pull/273#discussion_r1952721533

Reply:
Gonna resolve this one since the aforementioned conversation is now resolved